### PR TITLE
feature/SM-7109 upgrade tsoa

### PR DESCRIPTION
**Pull request by RichardEustace**
## Description

Please explain briefly the changes you made here.

## Coding Standards

Style guide: https://github.com/departmentfortransport/street-manager/wiki/Street-Manager-Style-Guide
Coding best practices: https://github.com/departmentfortransport/street-manager/wiki/Street-Manager-Coding-Best-Practices

## Checklist

- [ ] Branch named {feature|hotfix|task}/{SM-.*}
- [ ] If your pull request depends on any other, please link them
- [ ] Changes approved by your team
- [ ] Changes approved by another team
- [ ] API definitions updated
- [ ] Commit messages are meaningful
- [ ] Add `DO NOT MERGE` if you want to postpone merge

### CODE REVIEW

File: src/interfaces/workHistoryResponse.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export interface WorkHistorySummaryResponse {
   username: string
   event: AuditEvent
   event_string: string
-  object_reference?: string
+  object_reference?: string| null
```
Comment (cpaul92) [inline src/interfaces/workHistoryResponse.ts:15]: How come only some response properties have `| null`?

Comment (RichardEustace) [reply src/interfaces/workHistoryResponse.ts]: I meant to add these back in before Code review but rushed it.  They are there now.

File: src/interfaces/siteSummaryResponse.ts [hunk 2]
```diff
@@ -2,15 +2,15 @@ import { ReinstatementStatusResponse, ReinstatementTypeResponse } from './refere
 import { SiteDetails } from './siteDetails'
 
 export interface SiteSummaryResponse extends SiteDetails {
-  permit_reference_number?: string
-  depth?: number
-  length?: number
-  width?: number
+  permit_reference_number?: string| null | null
```
Comment (cpaul92) [inline src/interfaces/siteSummaryResponse.ts:5]:
```suggestion
  permit_reference_number?: string| null
```

File: src/interfaces/forwardPlanCreateRequest.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export interface ForwardPlanCreateRequest extends BaseWorkCreateRequest {
   project_reference_number?: string
   traffic_management_type?: TrafficManagementType
   /** Max length 500 characters */
-  additional_info?: string,
+  additional_info?: string
```
Comment (cpaul92) [inline src/interfaces/forwardPlanCreateRequest.ts:21]: This file seems to have been missed

Comment (ccurran01) [reply src/interfaces/forwardPlanCreateRequest.ts]: Did you run build after updating this file? I don't seem to be able to find the .d.ts file

Comment (RichardEustace) [reply src/interfaces/forwardPlanCreateRequest.ts]: Nope good spot, done now

**Review by cpaul92 — APPROVED** [review]

**Review by ccurran01 — APPROVED** [review]

File: src/interfaces/referenceTypes.ts [hunk 4]
```diff
@@ -786,7 +787,8 @@ export enum ASDPeriodicityCode {
   sunday_only = 13,
   public_and_bank_holidays = 14,
   continuous_62 = 15,
-  special_arrangements_63 = 16
+  special_arrangements_63 = 16,
+  null
```
Comment (cpaul92) [inline src/interfaces/referenceTypes.ts:791]: @RichardEustace I assume this is hangover from some previous testing and this null can be removed?

File: src/interfaces/referenceTypes.ts [hunk 5]
```diff
@@ -52,7 +52,8 @@ export enum CollaborationType {
   trench_sharing_primary_promoter = 'trench_sharing_primary_promoter',
   trench_sharing_secondary_promoter = 'trench_sharing_secondary_promoter',
   sharing_traffic_management_only = 'sharing_traffic_management_only',
-  other = 'other'
+  other = 'other',
+  null = 'null'
```
Comment (cpaul92) [inline src/interfaces/referenceTypes.ts:56]: @RichardEustace I assume this is hangover from some previous testing and this null can be removed?

File: src/interfaces/referenceTypes.ts (hunk 4 above)
Comment (RichardEustace) [reply src/interfaces/referenceTypes.ts:791]: Whoops, yeah

**Pull request merged**